lms/djangoapps/courseware/models.py

```diff
@@ -248,3 +255,113 @@ class Meta(object):   # pylint: disable=missing-docstring
 
     field = models.CharField(max_length=255)
     value = models.TextField(default='null')
+
+
+SCORE_CHANGED = Signal(
+    providing_args=[
+        'points_possible',  # Maximum score available for the exercise
+        'points_earned',   # Score obtained by the user
+        'user',  # User object representing the student
+        'course',  # CourseLocator object representing the course
+        'usage'  # BlockUsageLocator indicating the courseware instance
+    ]
+)
+
+
+@receiver(score_set)
+def submissions_score_set_handler(sender, **kwargs):  # pylint: disable=unused-argument
+    """
+    Consume the score_set signal defined in the Submissions API, and convert it
+    to a SCORE_CHANGED signal defined in this module. Converts the unicode keys
+    for user, course and item into the standard representation for the
+    SCORE_CHANGED signal.
+
+    This method expects that the kwargs dictionary will contain the following
+    values (See the definition of score_set):
+      - points_possible: integer,
+      - points_earned: integer,
+      - anonymous_user_id': unicode,
+      - course_id: unicode,
+      - item_id: unicode
+    """
+    user = None
+    course = None
+    usage = None
+    points_possible = kwargs.get('points_possible', None)
+    points_earned = kwargs.get('points_earned', None)
+
+    if 'anonymous_user_id' in kwargs:
+        user = user_by_anonymous_id(kwargs.get('anonymous_user_id'))
+    try:
+        if 'course_id' in kwargs:
+            course = CourseLocator.from_string(kwargs.get('course_id'))
```

Comment:
Please use `CourseKey.from_string()` and not `CourseLocator.from_string()`. The CourseKey will handle old and new style course IDs.